File: src/main/scala/com/badlogic/gdx/utils/viewport/FitViewport.scala
package com.badlogic.gdx.utils.viewport


import com.badlogic.gdx.graphics.Camera

import scala.scalajs.js
import scala.scalajs.js.annotation.JSName

/** A ScalingViewport that uses {@link Scaling#fit} so it keeps the aspect ratio by scaling the world up to fit the screen, adding
  * black bars (letterboxing) for the remaining space.
  * @author Daniel Holderbaum
  * @author Nathan Sweet */
@JSName("gdx.utils.viewport.FitViewport")
@js.native
class FitViewport extends ScalingViewport {
  /** Creates a new viewport using a new {@link OrthographicCamera}. */
  def this(worldWidth: Double, worldHeight: Double, camera: Camera = ???) = this()
}

File: src/main/libgdx/gdx/utils/viewport/FitViewport.js
gdx.utils.viewport.FitViewport = (function() {

    var Scaling = gdx.utils.Scaling;
    var ScalingViewport = gdx.utils.viewport.ScalingViewport;
    /**
     * @JSName("gdx.utils.viewport.FitViewport")
     */
    return class FitViewport extends ScalingViewport {
        constructor(worldWidth, worldHeight, camera) {
            super(Scaling.fit, worldWidth, worldHeight, camera);
        }
    }

}());
